modules/blockbuilder/blockbuilder.go

```diff
@@ -111,19 +111,16 @@ func (b *BlockBuilder) starting(ctx context.Context) (err error) {
 }
 
 func (b *BlockBuilder) running(ctx context.Context) error {
-	cycleEndTime := cycleEndAtStartup(time.Now(), b.cfg.ConsumeCycleDuration)
-	err := b.consumeCycle(ctx, cycleEndTime)
-	if err != nil {
-		return fmt.Errorf("failed to consume cycle: %w", err)
-	}
```

Comment:
It's this part: Fix block-builder startup error where the block builder caused a module failure before the ingester had time to create the partition. This was because the block-builder was treating a consumeCycle error as a hard module failure. Now it keeps retrying on a timer.

Returning an error from `running()` stops the program. A polling failure shouldn't cause the pod to terminate, it should enter the loop and continue trying, similar to compactor/etc.

Reply:
Makes sense, thanks

Reply:
I was again around this code, deleting `cycleEndAtStartup` could be problematic.

`nextCycleEnd` will look at the next cycle in the future (eg. at 10:02 with an interval of 15min, next cycle is 10:15). At startup it should look at the first cycle in the past so the data from 10:00 to 10:02 is not lost—first cycle would be 10:00 to 10:15.